### PR TITLE
relaxed factoid names to include + and -

### PR DESCRIPTION
**Pull request by quicquid**
the detection of the command is now separate from the factoid. just the pattern re_factoid needs to be adapted now, if we want to include more.

### CODE REVIEW

File: src/core/Plugin_factoids.ml [hunk 1]
```diff
@@ -175,15 +192,15 @@ let search tokens (fcs:t): string list =
...
-    | [] -> ""
-    | _ ->
-      let _, fact = Rand_distrib.uniform l |> Rand_distrib.run in
-      let msg_val = match fact.value with
-        | StrList [] -> assert false
-        | StrList l -> Rand_distrib.uniform l |> Rand_distrib.run
-        | Int i -> string_of_int i
-      in
-      Printf.sprintf "!%s: %s" fact.key msg_val
+  | [] -> ""
```
Comment (c-cube) [inline src/core/Plugin_factoids.ml]: please do not reindent like this :p 
ocp-indent should preserve the old indentation, I believe

File: src/core/Plugin_factoids.ml [hunk 2]
```diff
@@ -46,12 +46,16 @@ let mk_factoid key value =
...
 
-let re_set = Re_perl.compile_pat "^![ ]*([^!=+ :-]+)[ ]*=(.*)$"
-let re_set_force = Re_perl.compile_pat "^![ ]*([^!=+ :-]+)[ ]*:=(.*)$"
-let re_append = Re_perl.compile_pat "^![ ]*([^!=+ :-]+)[ ]*\\+=(.*)$"
-let re_get = Re_perl.compile_pat "^![ ]*([^!=+ :-]+)[ ]*$"
-let re_incr = Re_perl.compile_pat "^![ ]*([^!=+ :-]+)[ ]*\\+\\+[ ]*$"
-let re_decr = Re_perl.compile_pat "^![ ]*([^!=+ :-]+)[ ]*--[ ]*$"
+(* joins the result of Re_perl.split_full back together*)
+let group_join =
+  let rec aux acc = function
```
Comment (c-cube) [inline src/core/Plugin_factoids.ml]: not too important, but I'd use a buffer for faster concatenation…

File: src/core/Plugin_factoids.ml (hunk 1 above)
Comment (quicquid) [inline src/core/Plugin_factoids.ml]: For some reason, ocp-indent changed the identation on the lower block - I was also surprised. I'll restore the old indentation.

**Comment by quicquid** [comment]
I changed to buffers and removed the indentation issue.

**Comment by c-cube** [comment]
Thanks!

**Pull request merged**